Cargo.toml

```diff
@@ -66,3 +66,4 @@ tracing = { version = "0.1", default-features = false }
 tracing-core = { version = "0.1", default-features = false }
 tracing-subscriber = { version = "0.3", default-features = false }
 url = { version = "2.2", default-features = false }
+urlencoding = { version = "2.1.3" }
```

Comment:
It seems this repo is unmaintained(last commit indicates the status is `looking for maintainer`). Would be the logic simple enough to implement on our own

Reply:
Makes sense, it isn't too hard to implement. `urlencoding` seems to be quite widely used according to [statistics here](https://crates.io/crates/urlencoding), I didn't see it's unmaintained.

Let me know your preference, I can add an encoding-implementation if that's desired.

Reply:
> I think if it's easy we should just implement it ourselves. I want to avoid depending on creates that doesn't have active maintainers

+1

Reply:
Sounds good, I removed the dependency and added an inline implementation.

Reply:
`urlencoding`  is also used in the `opentelemetry` crate for the Baggage